pkg/platform/runtime/target.go

```diff
@@ -73,14 +80,12 @@ func (c *CustomTarget) OnlyUseCache() bool {
 	return c.commitUUID == ""
 }
 
-func ProjectDirToTargetDir(projectDir, cacheDir string) string {
-	projectDirDirty := filepath.Dir(projectDir)
-	projectDir, err := fileutils.ResolveUniquePath(projectDirDirty)
+func ProjectDirToTargetDir(projectDir, cacheDir string) (string, error) {
+	resolvedDir, err := fileutils.ResolveUniquePath(projectDir)
 	if err != nil {
-		logging.Error("Could not resolve unique path for projectDir: %s, error: %s", projectDir, err.Error())
-		projectDir = projectDirDirty
+		return "", locale.WrapError(err, "err_target_resolve_dir", "Could not resolve project directory")
```

Comment:
hmm... unless, the `projectDir` is the empty string, it should be safe to proceed with that string, shouldn't it?

The function doesn't seem to [ever have returned an error](https://rollbar.com/activestate/all/items/?sort=%5Bobject%20Object%5D&status=active&enc_query=PaPqSdMv2PrGOy8peB3IzqJj5jOl8MdnmGGo0SqqdLBbsoFs7oZ8NzfUGZVDYZOSyhy%2F%2BdYFFLQhWPcq%2Bml8PA%3D%3D&date_from=&date_to=&environments=beta&environments=master&environments=release&environments=unstable&environments=unstable-install.ps1&environments=unstable-install.sh&environments=unstable-msi-silent&environments=unstable-msi-ui&environments=unstable-unknown&environments=unstable-%EF%BF%BD%EF%BF%BDi%00n%00s%00t%00a%00l%00l%00.%00p%00s%001%00%20%00%20%00&activated_to=&timezone=America%2FLos_Angeles&framework=&levels=10&levels=20&levels=30&levels=40&levels=50&activated_from=&offset=0&query=Could%20not%20resolve%20unique%20path%20for%20projectDir&assigned_user=&date_filtering=seen&projects=291966).  So, it shouldn't be necessary to change that behavior IMO...

Reply:
Fair point. I've reverted the returning of an error but this function has still changed slightly